include/swift/AST/DiagnosticsSema.def

```diff
@@ -1506,6 +1506,11 @@ ERROR(protocol_composition_not_protocol,none,
       "non-protocol type %0 cannot be used within a protocol composition", (Type))
 ERROR(objc_protocol_inherits_non_objc_protocol,none,
       "@objc protocol %0 cannot refine non-@objc protocol %1", (Type, Type))
+WARNING(protocol_composition_with_postfix,none,
+        "protocol composition with postfix '%0' is ambiguous "
+        "and will be rejected in future version of Swift", (StringRef))
+NOTE(protocol_composition_with_postfix_suppress_warning,none,
+     "wrap the composition in parentheses to suppress this warning", ())
```

Comment:
Makes sense.
`protocol composition with postfix '.Type' is ambiguous and will be rejected in future version of Swift; wrap the composition in parentheses to disambiguate` is a little bit too long?
```
test.swift:3:23: warning: protocol composition with postfix '.Type' is ambiguous and will be rejected in future version of Swift; wrap the composition in parentheses to disambiguate
typealias P = P1 & P2.Type
              ~~~~~~~~^~~~
              (      )
```